### DISCUSSION TITLE
Help with OpenWRT and Strongswan

### DISCUSSION THREAD

**Discussion by jamnab1** [strongswan/strongswan#1434] [Q&A]
Hello.

I'm trying to set up OpenWRT to route all LAN traffic to a VPN service, using Strongswan. 

My configuration is pretty simple:

```
connections {
        vpn-0 {
                version = 2
                remote_addrs = 1.2.3.4
                local_addrs = 192.168.2.1
                vips = 0.0.0.0
                dpd_delay = 300s
                children {
                        test {
                                start_action = none
                                dpd_action = restart
                                mode = tunnel
                                remote_ts = 0.0.0.0/0
                        }

                }
                local-0 {
                        auth = eap-mschapv2
                        eap_id = userid

                }
                remote-0 {
                        auth = pubkey
                        id = %any
                }
        }
}
```

The status I get is something like this:
```
Connections:
        vpn-0:  192.168.2.1...1.2.3.4  IKEv2, dpddelay=300s
        vpn-0:   local:  uses EAP_MSCHAPV2 authentication with EAP identity 'userid'
        vpn-0:   remote: [%any] uses public key authentication
         test:   child:  dynamic === 0.0.0.0/0 TUNNEL, dpdaction=restart
Security Associations (1 up, 0 connecting):
        vpn-0[1]: ESTABLISHED 23 minutes ago, 192.168.2.1[192.168.2.1]...1.2.3.4[CN=vpn.com]
        vpn-0[1]: IKEv2 SPIs: aaaaaaaae8ba0127_i* aaaaaaaa55ec1ed5_r, rekeying in 3 hours
        vpn-0[1]: IKE proposal: AES_GCM_16_128/PRF_HMAC_SHA2_256/CURVE_25519
         test{1}:  INSTALLED, TUNNEL, reqid 1, ESP in UDP SPIs: aaaaaab2_i aaaaaafc_o
         test{1}:  AES_CBC_128/HMAC_SHA2_256_128, 78139 bytes_i (635 pkts, 13s ago), 88388 bytes_o (1024 pkts, 2s ago), rekeying in 33 minutes
         test{1}:  10.15.3.151/32 === 0.0.0.0/0
```

I set up a default VTI device and I can see incoming packets on that interface using tcpdump. However, I don't see how to route the LAN to the adapter.

I'm mostly using defaults. The charon config is all defaults. I don't understand this system very well and I would appreciate any help. For example, I'm not sure how it knows to use the VTI device.

Thanks!

**Comment by tobiasbrunner** [comment]
> I set up a default VTI device and I can see incoming packets on that interface using tcpdump.

That seems weird because I don't see any marks set in your config that would associate IPsec traffic with a VTI interface. Just know that using VTIs is NOT the default. See [here](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html) for details on route-based VPNs. However, note that when using virtual IPs, you will need an updown script that installs a NAT rule that maps your local traffic to the virtual IP so it matches the IPsec policy (whether you use a VTI or not doesn't matter).

**Reply by jamnab1** [reply]
Thanks for the reply. I remember reading that the standard approach doesn't use VTIs. I would rather use that method but I don't really understand it. Also, OpenWRT uses fw4 rather than iptables. Is that a potential issue?

I had to build an updown script, as you indicated and now things mostly work. NAT works when originating from the LAN. However, port forwards aren't getting through.

Maybe I should try reworking this as rules, rather than using the route method. I don't know if there's any advantage to using routing. What do you think?

**Reply by tobiasbrunner** [reply]
> However, port forwards aren't getting through.

What port forwards?

> Maybe I should try reworking this as rules, rather than using the route method. 

What do you mean?
